flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java

```diff
@@ -283,6 +286,76 @@ void testToLiteralMap() {
                 )));
   }
 
+  @Test
+  public void testToSdkBindingDataMap() {
+    AutoValueInput input =
+        createAutoValueInput(
+            /* i= */ 42L,
+            /* f= */ 42.0d,
+            /* s= */ "42",
+            /* b= */ false,
+            /* t= */ Instant.ofEpochSecond(42, 1),
+            /* d= */ Duration.ofSeconds(1, 42),
+            /// * blob= */ blob,
+            /* l= */ List.of("foo"),
+            /* m= */ Map.of("marco", "polo"),
+            /* ll= */ List.of(List.of("foo", "bar"), List.of("a", "b", "c")),
+            /* lm= */ List.of(Map.of("A", "a", "B", "b"), Map.of("a", "A", "b", "B")),
+            /* ml= */ Map.of("frodo", List.of("baggins", "bolson")),
+            /* mm= */ Map.of(
+                "math", Map.of("pi", "3.14", "e", "2.72"), "pokemon", Map.of("ash", "pikachu")));
+
+    Map<String, SdkBindingData<?>> sdkBindingDataMap =
+        JacksonSdkType.of(AutoValueInput.class).toSdkBindingMap(input);
+
+    Map<String, SdkBindingData<?>> expected = new HashMap<>();
+    expected.put("i", input.i());
+    expected.put("f", input.f());
+    expected.put("s", input.s());
+    expected.put("b", input.b());
+    expected.put("t", input.t());
+    expected.put("d", input.d());
+    expected.put("l", input.l());
+    expected.put("m", input.m());
+    expected.put("ll", input.ll());
+    expected.put("lm", input.lm());
+    expected.put("ml", input.ml());
+    expected.put("mm", input.mm());
+
+    assertEquals(expected, sdkBindingDataMap);
+  }
+
+  @Test
+  public void testToSdkBindingDataMapJsonProperties() {
+
+    JsonPropertyClassInput input =
+        new JsonPropertyClassInput(
+            SdkBindingData.ofString("test"), SdkBindingData.ofString("name"));
+    System.out.println(JacksonSdkType.of(JsonPropertyClassInput.class).getVariableMap());
+
+    Map<String, SdkBindingData<?>> sdkBindingDataMap =
+        JacksonSdkType.of(JsonPropertyClassInput.class).toSdkBindingMap(input);
+
+    Map<String, SdkBindingData<?>> expected = new HashMap<>();
+    expected.put("test", input.test);
+    expected.put("name", input.otherTest);
```

Comment:
```suggestion
    var expected = Map.of("test", input.test, "name", input.otherTest);
```
The Sdk was written when we were supporting Java8 but now the minimum in Java11, so lets take the opportunity to modernize code when we can